Combat Patrol game mode - TGMC vs SOM

Lumipharon opened the pull request:
<!-- ***STOP!***  Read this: If this is not a PR ready for review and merge or WIP, open it as a draft PR, using the arrow next to 'Create Pull Request'>

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
### **Combat Patrol**
This game mode is a simple team death match. with gameplay somewhat streamlined to get your unga on faster.

This gamemode is mainly aimmed at lowpop, and is in no way looking to supplant HvX, but hopefully it can add a bit of variety every now and then. If it doesn't end up being hot garbage, more detailed game mode can be built with much greater ease in the future, now that SOM has been more fleshed out.

This is also basically a minimum viable product. I will of course pile in more loadouts and tweak thing based on feedback, so please do provide it.

### **Round start**
Instead of spawning shipside, you spawn in a patrol base, with only Quick Vendors. These have a selection of premade loadouts for you to choose from, based on your role. You only get one choice so choose wisely!
![image](https://user-images.githubusercontent.com/7869430/178246064-b851bcf1-23ba-4306-b9e4-8cf9067129b4.png)

Shout out to Dagger for making actual nice looking patrol bases.

After your 5 second prep time you're good to go, but you cannot deploy under shutters go up (after 5 minutes, currently). Each side has two sets of patrol entry points, that exit at roughly opposite sides of the map, with the other team taking the other two corners. This means you can choose to split your forces, or stick into a single unga ball.
**This is a one way trip**, once you've left base you're in the field for the rest of the round. Make sure you're properly ready to leave, and you probably don't want to take one exit when the rest of your team chooses to take the other.
![image](https://user-images.githubusercontent.com/7869430/178246482-6a01669a-1623-4121-9599-d833d2ece0dc.png)


### **Groundside**
Once shutters go up, the patrol lasts for up to 35 minutes, and the name of the game is to hunt down your enemy and kill them all. This game mode is set during the day, so everything is lit, although caves and interiors are a bit darker still.

**There is no medivac**
No medivac means you can't get sent off for surgery if you get fucked up. Either a medic does what they can, or if you're lucky you can take over groundside medbay and use that. Potentially a game changer if people start racking up lots of broken bones or organ damage.

Although there's no firesupport like CAS or Mortars, You do have the equipment to fortify if you wish, and marines especially have some gear that can make that a bit more effective, especially since most of their weapons have aim mode. Marines also have a wider variety of weapons, such as deployable machineguns, recoilless rifles etc.

On the SOM side, they have volkite for more raw damage (but no aim mode) and generally less flexibility of kit, so will likely have to rely on being aggressive to carry the day, as well as trying to avoid areas where they can get funneled and murdered when they can't all fire effectively.

### **Win conditions**
The game ends when one side is completely wiped from the ground map or the timer runs out. The timer is visible in the game tab.

- Major victory for wiping the enemy
- Minor victory for scoring more kills at round end
- Draw for equal kills or if you manage a mutual KO somehow

Respawns occur in waves. Every 10 minutes after the shutters drop, anyone dead will be able to respawn. When respawning you cannot change faction, but going perma (including respawning) frees up your job slot.
Points are based off the number of dead at the end of the round, so unga respawning just to die again is going to help the enemy, while actually reviving people and staying alive helps your team.

### **Sons of Mars**
![image](https://user-images.githubusercontent.com/7869430/177998429-9906e80f-8ece-4b47-bddf-32a5d13ecf60.png)


The Sons of Mars take the stage as a fully fledged faction.

They have squads (Zulu, Yankee, Xray, Whiskey), radio channels, minimap icons, all that fun jazz. Currently only squad roles, but more may come in the future.

Mission critical update: Their uniforms can now roll their sleeve up.
![image](https://user-images.githubusercontent.com/7869430/178246858-033da09d-f7b9-4066-94b5-bb9f6ce888fe.png)


### **Equipment**
Loadouts have been vaguely balanced for HvH. It will almost certainly not be well balanced to start with however. :^)

There is a bunch of new kit made specifically for HvH, some of it has been added in previous PR's but listed here for relevance. There will also be more stuff coming in the future as well.

**Grenades**
![image](https://user-images.githubusercontent.com/7869430/178248844-59f18d03-ecd0-4142-b50c-e71ccc5c85d2.png)

Stun grenades: 
**Does not cause any hard stuns**
The closer you are to the detonation, the more powerful the effect. Applies a very short blind, as well as deafening, staggering and slowing the target. Used by both sides.

Acid smoke grenades: 
TGMC issue, derived from xenomorphs. Pops a cloud of acid like a boiler.

Neurotoxin smoke grenade: 
Ditto above, but with neurotoxin.

Satrapine smoke grenade: 
Pops a cloud of nerve agent, causing escalating pain, and later lossbreath and coughing blood. Also drains common painkillers like tramadol slowly. Used by the SOM.

**New SOM weapons**
Shout out to Iamtheguy for the nice weapon sprites (except for my shitty volkite pistol sprite to go with my other volkite coder sprites).

**V-11 pistol:**
![image](https://user-images.githubusercontent.com/7869430/178248139-951d7bba-8a7a-42be-9263-6af85a06c601.png)

9mm side arm, 18 round mag with AP pistol rounds. Comparable to the 88 Mod 4, probably worse than a P-23 overall.

**V-21 submachinegun:**
![image](https://user-images.githubusercontent.com/7869430/178247832-11057c5e-e5ae-437b-bbdf-41d00eb828cd.png)

50 round SMG, comparable but slightly worse than a P90 one handed. However has a higher rate of fire mode (toggled via autoburst), giving you 0.1 fire delay instead of 0.15, at the cost of significantly degraded accuracy and scatter. Ineffective one handed, but deadly when wielded, although you'll burn through ammo quick.

**V-31 assault rifle:**
![image](https://user-images.githubusercontent.com/7869430/178247704-886c6e12-eb0e-4238-a11b-b5ccaf51d5a3.png)

The mainstay rifle of the SOM (the volkite weapons are their shiny new high end stuff). Has a 50 round magazine designed mainly for burst fire, with damage output somewhere between a AR-12 and AR-18, (so worse than an AR-18 at close range, and worse then the AR-12 at longer range, due to worse falloff). 
Has an integrated grenade launcher, which is a low velocity railgun (no need for propellant means more space in a small package of the payload), with a variety of grenade types.

Firstly, the GL is loaded like a masterkey (3 shots, and you can mix and match what's loaded), and the grenades have some specific behavior.
The grenades have to prime after being launched. If you hit something within 2 tiles, the grenade will fail to detonate, effectively wasting it (mad points if you kill someone with it though). If it gets past that range, the grenade detonates mid air, and releases it's payload, in a similar way to the GL-54.

![image](https://user-images.githubusercontent.com/7869430/178246978-a47abfa2-4b94-464d-90c9-cd67c199818a.png)

From left to right:
_Airburst:_ 
A cone of mob piercing shrapnel. Causes reasonable damage, but decent stagger and slow.
_Dragon's breath:_ 
A cone of mob piercing, incendiary shrapnel. Less damage and very low stagger/slow, but sets mobs on fire with a chance to ignite turfs.
_Cluster grenade:_ 
Releases a cloud of bomblets in a cone pattern. Causes reasonable damage and decent stagger/slow over a fairly large area, but less effective against a single or tighly groupped targets compared to airburst.
_Smokescreen:_ 
Releases a number of smoke bomblets that will create a literal smokescreen, or otherwise explode when they impact something.

**V-51 combat shotgun**
![image](https://user-images.githubusercontent.com/7869430/178247945-3248e7bc-6c9d-4c49-8121-5dc835d44e66.png)

Semi auto shotgun, comparable to the SH-39, slightly slower firing but more damage. Cannot be used one handed. Uses normal shotgun ammo.

**VX-12 volkite serpenta**
![image](https://user-images.githubusercontent.com/7869430/178248246-9d3bc265-6ff6-42f2-9f09-dd2a01333a6a.png)

...Not to scale (I blame Kuro for corrupting my mind with giant miniguns).
A volkite pistol, normally seen on SOM officers and some NCO's. Extremely potent, but a low capacity.

**COPE sentry**
![image](https://user-images.githubusercontent.com/7869430/178481770-c3ccc7f6-5734-4197-81ce-8de3f2cc31bb.png)

The COPE is a sentry turret that is deployed like a grenade, making it a bit easier to use. It also hugs the ground so isn't a dense object and is a bit harder to hit. 
Packs a punch with a quick firing volkite weapon and a recharging battery, SOM engineers can only get one of these however.

And a bunch of miscellaneous stuff including:
SOM specific storage items like satchels, pouches, etc. All in classy SOM leather.
Riot shields no longer have 100 energy or rad armor (making the user invulnerable to volkite outside of deflagrate!) , but have a bit more laser armor.

### **Preferences**
Adds a Preferred squad - SOM selection as well as the new SOM jobs to job preferences.
![image](https://user-images.githubusercontent.com/7869430/178245820-aa61a9d2-9d7c-44b1-8e23-7dff1dee5f79.png)


Will presumably need a config update to add the combat patrol base shipmap to the roster, but all ground map (other than Icarus) support the game mode.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Combat Patrol - a new gamemode pitting the TGMC vs SOM in a TDM
add: Fleshed out the SOM as a proper faction, with their own squads and related features
add: A variety of new SOM weaponry has been introduced to flesh out their gear options
balance: Volkite weapons have been tuned down slightly
balance: SOM armour has been tuned down slightly
fix: Marine shield no longer has 100 energy armor
fix: Volkite deflagrate correctly takes damage fall off into account
fix: Volkite deflagrate soft CC max range fixed
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Luxyboo commented:
hopefully this is a better CW

revelation-8468 commented:
Are there SOM combat robots?

Or is it humie only for either side?

Lumipharon commented:
> Are there SOM combat robots?
> 
> Or is it humie only for either side?

Human only for now.

>I'm afraid that the stun grenades, even if heavily restricted will become the meta. I don't think anyone wants to experience the fresh hell that is synths with flashbangs in EORD.

Stun grenades don't hard stun, and the blind is literally less that a second. Also players normally only get 2.

Al-1ce commented:
> Stun grenades don't hard stun, and the blind is literally less that a second. Also players normally only get 2.

That might be fine then. With that said, the pepperball gun still takes the cake for HvH meta.

Lumipharon commented:
Mapping is actually cursed.

RipGrayson commented:
Can confirm here, one of the maps here might be genuinely haunted and I'm still working through it with Lumi.

Lumipharon commented:
Map diff why you do this to me, I just updated the shutters. Guh.

Lumipharon commented:
Yeah I did that initially but then I was getting those spooky issues, I'll change it back tomorrow.

Lumipharon commented:
@RipGrayson updated. 

Someone that's not me can make a pretty version of the patrol base in the future, I just went for the ugly bare essentials.

TiviPlus commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -0,0 +1,188 @@
+	to_chat(world, "<b>The current game mode is - Combat Patrol!</b>")
+	to_chat(world, "<b>The TGMC and SOM both lay claim to this planet. Across contested areas, small combat patrols frequently clash in their bid to enforce their respective claims. Seek and destroy any hostiles you encounter, good hunting!</b>")
+	to_chat(world, "<b>WIP, report bugs on the github!</b>")
+
+
```
```suggestion
```

TiviPlus commented on code/datums/jobs/job/sonsofmars.dm:
```diff
@@ -11,21 +10,222 @@
+/datum/outfit/job/som/squad/veteran
+	name = "SOM Veteran"
+	jobtype = /datum/job/som/squad/veteran
+
+	id = /obj/item/card/id/dogtag/som
+
```
```suggestion
	id = /obj/item/card/id/dogtag/som
```

TiviPlus commented on code/datums/jobs/job/sonsofmars.dm:
```diff
@@ -11,21 +10,222 @@
+	jobtype = /datum/job/som/squad/leader
+
+	id = /obj/item/card/id/dogtag/som
+
+
+//ERT roles
+
```
```suggestion

//ERT roles
```

TiviPlus commented on code/game/objects/items/explosives/plastique.dm:
```diff
@@ -18,7 +18,7 @@
 	var/datum/effect_system/smoke_spread/smoketype = /datum/effect_system/smoke_spread/bad
 	/// radius this smoke will encompass
-	var/smokeradius = 2
+	var/smokeradius = 1
 
```
???

Lumipharon replied:
>C4 now creates less smoke, for a much shorter duration. It was kinda shitty for breaching because you'd have to wait for days for the smoke to clear, or charge through blind.

Basically if you want to actually use it to breach a building with hostiles in it, you have to wait 2 years for a cloud of smoke to clear, which makes C4 really bad.

Lumipharon replied:
removed.

TiviPlus commented on code/game/objects/items/explosives/plastique.dm:
```diff
@@ -113,7 +113,7 @@
 	explosion(plant_target, 0, 0, 1, 0, 0, 1, 0, 1)
 	var/datum/effect_system/smoke_spread/smoke = new smoketype()
-	smoke.set_up(smokeradius, loc, 11)
+	smoke.set_up(smokeradius, loc, 2)
 	smoke.start()
```
???

Lumipharon replied:
See above

Lumipharon replied:
removed.

TiviPlus commented on code/game/objects/items/radio/headset.dm:
```diff
@@ -792,3 +807,127 @@ GLOBAL_LIST_INIT(channel_tokens, list(
+	minimap_type = /datum/action/minimap/som
+
+/obj/item/radio/headset/mainship/som/Initialize(mapload, datum/squad/squad, rank)
+	if(squad)
+		icon_state = "headset_marine_[lowertext(squad.name)]"
```
early returns

Lumipharon replied:
ok

TiviPlus commented on code/game/objects/structures/patrol_points.dm:
```diff
@@ -0,0 +1,51 @@
+	var/obj/effect/landmark/patrol_point/linked_point = null
+
+/obj/structure/patrol_point/Initialize()
+	. = ..()
+
```
```suggestion
	..()
```

Lumipharon replied:
ok

TiviPlus commented on code/game/objects/structures/patrol_points.dm:
```diff
@@ -0,0 +1,51 @@
+/obj/structure/patrol_point/LateInitialize()
+	. = ..()
+	create_link()
+
+/obj/structure/patrol_point/proc/create_link()
```
autodoc

Lumipharon replied:
ok

TiviPlus commented on code/game/verbs/who.dm:
```diff
@@ -65,7 +68,7 @@
 		msg += "<b>Total Players: [length(Lines)]</b>"
 		msg += "<br><b>Observers: [count_observers] (Non-Admin: [count_nonadmin_observers])</b>"
-		msg += "<br><b>Humans: [count_humans]</b> <b>(Marines: ~[count_marine_humans])</b> <b>(Infected: [count_infectedhumans])</b><br>"
+		msg += "<br><b>Humans: [count_humans]</b> <b>(Marines: ~[count_marine_humans])</b> <b>(Marines: ~[count_som_marine_humans])</b> <b>(Infected: [count_infectedhumans])</b><br>"
```
SOM Marines?

Lumipharon replied:
whoops

TiviPlus commented on code/modules/mob/camera/imaginary_friend.dm:
```diff
@@ -137,6 +137,8 @@
 	med_squad_mobhud ? H.add_hud_to(src) : H.remove_hud_from(src)
 	H = GLOB.huds[DATA_HUD_SQUAD_REBEL]
 	med_squad_mobhud ? H.add_hud_to(src) : H.remove_hud_from(src)
+	H = GLOB.huds[DATA_HUD_SQUAD_SOM] //what is this admemery?
+	med_squad_mobhud ? H.add_hud_to(src) : H.remove_hud_from(src)
```
?

Lumipharon replied:
I don't really know what this does because it's some admin thing, so I just copy pasted it for the SOM hud in case it's actually used for something.

TiviPlus commented on code/modules/mob/living/silicon/silicon.dm:
```diff
@@ -143,7 +143,12 @@
 			HUD_nbr = 2
 		if("Squad HUD")
-			H = GLOB.huds[faction == FACTION_TERRAGOV ? DATA_HUD_SQUAD_TERRAGOV :DATA_HUD_SQUAD_REBEL]
+			if(GLOB.huds[faction] == FACTION_TERRAGOV)
+				H = DATA_HUD_SQUAD_TERRAGOV
+			if(GLOB.huds[faction] == FACTION_TERRAGOV_REBEL)
+				H = DATA_HUD_SQUAD_REBEL
+			if(GLOB.huds[faction] == FACTION_SOM)
+				H = DATA_HUD_SQUAD_SOM
```
use else if

Lumipharon replied:
ok

TiviPlus commented on code/modules/projectiles/ammo_datums.dm:
```diff
@@ -1347,6 +1355,187 @@ datum/ammo/bullet/revolver/tp44
+	shrapnel_chance = 0
+	max_range = 6
+	bullet_color = COLOR_VERY_SOFT_YELLOW
+	var/datum/effect_system/smoke_spread/smoketype = /datum/effect_system/smoke_spread
+
+/datum/ammo/micro_rail_cluster/proc/detonate(turf/T, obj/projectile/P)
```
Autodoc

Lumipharon replied:
OK

TiviPlus commented on code/modules/projectiles/ammo_datums.dm:
```diff
@@ -1347,6 +1355,187 @@ datum/ammo/bullet/revolver/tp44
+		victim.visible_message(span_danger("[victim] is hit by the bomblet blast!"),
+			isxeno(victim) ? span_xenodanger("We are hit by the bomblet blast!") : span_highdanger("you are hit by the bomblet blast!"))
+		var/armor_block = victim.get_soft_armor("bomb")
+		victim.apply_damage(rand(5, 15), BRUTE, null, armor_block, updating_health = TRUE)
+		victim.apply_damage(rand(5, 15), BURN, null, armor_block, updating_health = TRUE)
```
```suggestion
		victim.apply_damage(rand(5, 15), BRUTE, null, armor_block, updating_health = FALSE)
		victim.apply_damage(rand(5, 15), BURN, null, armor_block, updating_health = TRUE)
```

TiviPlus replied:
Also this rand() is stinky

Lumipharon replied:
changed both

TiviPlus reviewed (changes requested):
First pass

DemoCrack commented:
Thank you, HVH gamemodes are pretty based.

novaepee commented:
I won't be surprised if people search for the nearest shotgun in planetside.

Lumipharon commented:
Both sides have shotguns, just not slugs.

Lumipharon commented:
Cleaned up the loadout file to make it a bit less bloated and more maintainable. Also added a unga AR18 loadout. I'll add more after some TM.

zzurge commented:
zased...so zased

novaepee commented:
Shipside map doesn't spawn into intended Combat Patrol map.

>be me
>choose SOM
>spawns in PoS
>oh oh
>guess I kill marines
>get loadout ASAP
>my usual marine equipment
>shoot marines
>hell breaks loose in prep
>I run to CIC
>log into announcement
>"you fuck up"
>admins restarts server
![image](https://user-images.githubusercontent.com/6610922/179367981-3518e33e-86b1-415a-832b-3583a2feb71a.png)

WrillWasTaken commented:
>be me
spawn in
eat protein bar
delicious
look to my left
literal warzone in prep

clearly bug free

novaepee commented:
SECOND ROUND OF COMBAT PATROL:

SOM minor at 12:14. How?
![image](https://user-images.githubusercontent.com/6610922/179368401-b077868c-1bbf-40f1-9f48-266ec88d01bf.png)

novaepee commented:
There is no storage inside SOM armor.

UzedPickle commented:
COPE turret (i think) is a bit too strong in a round end deathmatch my COPE killed like 7 pepole. a short hallway with a cope in it i think would be a bit too strong

novaepee commented:
![image](https://user-images.githubusercontent.com/6610922/179369066-58c9ac35-4bd6-4b60-99bb-bb3b1d9e8705.png)

Gamemode is Nuclear War but shipside is Combat Patrol

UzedPickle commented:
COPE turret buged out dunno why after i activated it kept flashing orange and never deployed.

bowlofcereal commented:
The gamemode forcibly ends the round at 12:13 and doesn't autopick the SOM patrol bases so admins have to force it or else the gamemode bugs and breaks the game, and SOM spawn with marines and prep turns into a bloodbath when they realize this. Also too many people have volkites, only SOM SGs should get a volkite. SOM were insta wiping marines like crazy with volkite spam. I untm'd it so it wouldn't cause any problems until this is tweaked.

blackdav123 commented:
wouldnt bother changing volkite availabilty until we get a full round with bugs fixed because we did have one round with marines having the upper hand

cope turret is a little op though with how fast it shoots

Lumipharon commented:
Ah the chorus of bugs. Sorry guys, certain things I can't test in dev with only one player.

OK so game timer (should) be fixed.
I'm looking now at the shipside vote thing, as that's supposed to be overridden.

Cope turret in particular, the idea is you kinda need to grenade it, or wall peak it etc as it is supposed to make a direct bum rush a terrible idea.

General balance shit I'll worry about when the game is actually running fine though.

Lumipharon commented:
OK timer and shipmode vote should both be fixed. Note shipmap vote is disabled when combat patrol is voted at round end, (and patrol base is force selected).
However via admemery or whatever else, shipmap can still be force called. So like, admins don't call a shipside vote after the gamemode vote ends.

UzedPickle commented:
i feel that there should be some objective. could be as simple as a fully stocked medbay or armory with op guns but currently there is no centralized thing/place to shoot at eachother over. its like when there are 3 xenos left just running and venting around the map and killing silos/nuke are not wincons its just a long search for things to shoot

Rexdyer1 commented:
Just finished playing a test round. Some thoughts for your consideration.
-first: Ice would be great for this gamemode
-second: having a defense and attack positions would be good. Having one side defend the colony while the other attacks. Have the defenders take up positions in the more heavily defended parts of the maps. Like the main colony underground in ice and the labs in big red.
Third: Maybe giving marines specialists and heavy weaponry to back them up to deal with the SOM volkite guns; some suggest reducing volkite, but that's kinda what makes SOM special, so it takes away the point of it if volkite isn't front and center.
Fourth: this is just a personal observation, but I think you have PMC's Oppose SOM instead of marines. PMC's have better gear(minus the SMG) than marines and seem to be a better fit for fighting SOM thematically.

blackdav123 commented:
I like how open ended the objectives are but adding some kind of objective that gives a bonus later into the round could be interesting, like an intel computer that drops in weapons or ammo or something.

UzedPickle commented:
In future tests you should isolate the gamemode itself from balanceing SOM vs TGMC as right now your not getting good fedback about the gamemode itself but rather a long list of SOM bugs and balance problems. to test if the gamemode itself is fun why not try TGMC vs TGMC or SOM vs SOM so you can test the gamemode itself rather then the balanceing overshadowing the gamemode feedback

Weedmaster77 commented:
might want to get rid of Pump shotty in armory and or slugs on all maps since you know why

Lumipharon commented:
Added AR-11 and SH-15 loadouts for marine/SL, put a limit on volkite loadouts for standard/engie/medic, and readded headsets exploding if you put on enemy ones.

This game mode is just supposed to be a pretty open ended TDM, mainly for crash pop. Going forwards we can totally add more objective focused modes quite easily, and I agree about many of the points raised.

I'll think about potential marine changes going forwards, but I want to see how it goes with less volkite spam.

XSlayer300 commented on code/__DEFINES/mode.dm:
```diff
@@ -69,6 +70,12 @@
+#define MODE_SOM_MARINE_MINOR "Marine Minor Victory"
+#define MODE_SOM_SOM_MAJOR "Sons of Mars Major Victory"
+#define MODE_SOM_SOM_MINOR "Sons of Mars Minor Victory"
+#define MODE_COMBAT_PATROL_DRAW "Draw: Both sides wiped out!"
```
```suggestion
#define MODE_COMBAT_PATROL_DRAW "DRAW: Both sides wiped out!"
```
consistancy.

Lumipharon replied:
Just gonna change it to the other draw message, since draw can occur without both sides completely wiping.

XSlayer300 commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -0,0 +1,189 @@
+/datum/game_mode/combat_patrol
+	name = "Combat patrol"
```
```suggestion
	name = "Combat Patrol"
```

Lumipharon replied:
This actually caused a funny bug... gotta double check if I use the lower case elsewhere first.

Lumipharon replied:
done.

BraveMole commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -0,0 +1,189 @@
+				area_to_lit.set_base_lighting(COLOR_WHITE, 75)
+			if(CEILING_DEEP_UNDERGROUND to CEILING_DEEP_UNDERGROUND_METAL)
+				area_to_lit.set_base_lighting(COLOR_WHITE, 25)
+	GLOB.join_as_robot_allowed = FALSE
```
no robot ? Why not btw ?

Lumipharon replied:
Because they ignore gas. And pain. And fire. And don't crit.
Basically I think they have an actually notable advantage in combat patrol, although it turns out this doesn't actually work, as atleast some people were able to join as combat robots today.

FosterGrime-Heymynameslouie commented:
I dont know if this is a bug or is intended but I vivdly remember peoples head being obliterated if they wear headsets that isint theres, I didnt see this happen whenver SoM uses TGMC head sets and vice versa, was this removed specifically for SoM and is comm sniffing and tacmap peeking a legitimate tactic?

BraveMole commented on code/game/objects/structures/patrol_points.dm:
```diff
@@ -0,0 +1,51 @@
+	anchored = TRUE
+	resistance_flags = RESIST_ALL
+	layer = LADDER_LAYER
+	//ID to link with associated exit point
```
autodoc is three /

Lumipharon replied:
whoops

BraveMole replied:
You resolved but didn't fix

Lumipharon replied:
done, whoops x2

BraveMole reviewed (changes requested):
pretty cool stuff

Lumipharon commented:
> I dont know if this is a bug or is intended but I vivdly remember peoples head being obliterated if they wear headsets that isint theres, I didnt see this happen whenver SoM uses TGMC head sets and vice versa, was this removed specifically for SoM and is comm sniffing and tacmap peeking a legitimate tactic?

That code got removed, but I've added it back with a tweak so should be working now/next time server updates.

Autisem commented:
some way to regain resupply my be neat

FosterGrime-Heymynameslouie commented:
![image](https://user-images.githubusercontent.com/59080633/179395934-37c95be1-fd01-40ea-b885-09034c17e179.png)

Supply drop like ressuply or having the SL's drop
the supply drop one might be fun since both sides can fight on it and the supply drop itself might give each team that secures an edge over the other.

fourtnaight

Autisem commented:
Consider ability for SL's to buy supplys. Maby by default put everyone into same squad. And most imporetntly

Add capture points.

Right now seems a mad dash for medbay and who ever is closest on spawning takes defense, maby doint the battle field style for 100 respawns and capturing capture points reduced enemy respawns until 1 side hits  0

XSlayer300 commented:
You cannot follow the SOM guys in ghost mode.

XSlayer300 commented:
Activating and getting the COPE sentry while it is primed will never activate it anymore, in addition you're stuck with a permanent light from the COPE that attaches to yourself.

BraveMole commented:
Som are not appearing in the follow mob ghost list, please fix

Losenis commented:
Breaks valhalla by making a lot of things undispensable from vendors. You might as well make this an extension of EORD with 0 respawn timers, the gamemode effectively kicks out xeno players and people who want to play vs xenos given they're the minority by how people who play marines are in the majority most if not allof the time.

Lumipharon commented:
@BraveMole Follow should now work for SOM, I forgot to update the actual UI.

Slight tweaks to volkite. They all have a bit less AP (this also means less deflagrate), charger and caliver have a bit more slowdown, and charger has 40 cap instead of 45, as well as slightly higher damage falloff.

Also cope turret has the same hp as other sentries now.

SOM standard, engie and medic charger loadouts now have magharnesses instead of reddot and laser sight. Minor tweak, but that's a chunk of accuracy removed, which will reduce the damage output slightly in some cases, especially at longer range.

Lumipharon commented:
Cope bug fixed - they'll now simply deactivate if they try to deploy when not on the floor.

Added some ~lore~ when you spawn, explaining what's going on and what the victory conditions are.

Lumipharon commented:
Added a shield and smg tankboi to marines. The shield has very good energy armour, so volkite basically bounces off except for when it triggers deflagrate, so should be able to face tank for the lads fairly well.

Weedmaster77 commented:
Shields seem like a bad idea maybe instead a armor for some of then role not marine reg that help against lasers. 

Also MG27 is kinda dumb the only thing close that SOM get is the COPE and its got a lot less range so unless you plan on giving SOM a plant-able MG  MG27 seem OP on defense

yyzsong commented:
SOM Vet weapons should probably need an SG skill check to be fired, given that they're the SOM version of the SG.

UzedPickle commented:
i think MG27 is fine. yes its unstopable defence but its so poor offensively that it felt fine to me. plus TGMC should also have cool toys rather just SOM with the cool shit

Weedmaster77 commented:
> i think MG27 is fine. yes its unstopable defence but its so poor offensively that it felt fine to me. plus TGMC should also have cool toys rather just SOM with the cool shit

That's my point is to much defense and your point of SOM getting cool toys makes not sense because yes they get all new weapon and marine are using normal stuff that they have had this mode was made and had to give SOM new stuff marine have everything and more that aright exists its not about giving marines cool new toys

Dueu commented:
The medical pouches that every loadout gets beside medic has advanced kits instead of gauze and ointment, way too slow to use for my taste and I remember something about the insta healing per kit applied has a halved effectiveness depending on your medical skill making more efficient using normal gauze or ointment if you aren’t medic or SL.

Dueu commented:
> Added a shield and smg tankboi to marines. The shield has very good energy armour, so volkite basically bounces off except for when it triggers deflagrate, so should be able to face tank for the lads fairly well.

Good addition, actually not overpowered due to shield and healing taking much longer to repair/heal than normal H vs X rounds. You need people actually pusing with you or pray you dont get chase if your shield is broken.

UzedPickle commented:
its getting quite good there are gamemode votes where its pretty close to nucular war now. also Round are a bit too long for my taste right now consider adding a end condition like "has double living playes then other team" or somthing as the action semes to die down over time.

carbous commented:
I like the idea but I think the ammount of respawn points should be made more numerous and the respawn timer should be made shorter to kind of create a constant battle, it would keep both side constantly engaged regardless of who's winning.

UzedPickle commented:
> I like the idea but I think the ammount of respawn points should be made more numerous and the respawn timer should be made shorter to kind of create a constant battle, it would keep both side constantly engaged regardless of who's winning.

i agree but be carefull that the timer dosent get too short and pepole start suicideing to avoid organ damage or IB that could become a problem

Lumipharon commented:
I'm probably going to figure out some wave spawning mechanism so it's not just a slow trickle of people that doesn't let you claw it back That may take some time however. I don't want to make respawn timer too low as it just becomes a mindless unga fest. Major victory is wiping the enemy for a reason.

Lumipharon commented:
Map diff fucks out sometimes due to the size of the pr. guh.

XSlayer300 commented:
> I'm probably going to figure out some wave spawning mechanism so it's not just a slow trickle of people that doesn't let you claw it back That may take some time however. I don't want to make respawn timer too low as it just becomes a mindless unga fest. Major victory is wiping the enemy for a reason.
> 
> Edit: (probably) fixed the end game timer not working correctly. It's a pain to try test alone. Also tweaked the timer to 35 minutes, as 40 is pretty long.

Could be something like the outpost airlocks closes after 4 minutes, then reopens at 14 minutes or something.

Weedmaster77 commented:
Please for the love of god make CP pop locked to 30 and admin pick its taking times where pop is high enough and just trashing it when people want to play xenos vs marines not HvH.  Or even make is so it can be played more then twice in a row after 30 pop if that something that can be done. 

Feel like its taking over the server and making it a HvH server not a Xenos Vs Marines  IMO

Alexeivichi commented:
Gamemode voting before the round starts for combat patrol while the round start is delayed seems to make the server think the round has ended, and causes it to enter and infinitely loop of gamemode + shipside votes by the server, followed by restarting after. Can you investigate this?

BraveMole commented on code/controllers/subsystem/vote.dm:
```diff
@@ -142,13 +142,13 @@ SUBSYSTEM_DEF(vote)
 			if(GLOB.master_mode != .)
+				SSticker.save_mode(.)
 				var/datum/game_mode/current_gamemode = config.pick_mode(GLOB.master_mode)
 				if((current_gamemode.flags_round_type & MODE_SPECIFIC_SHIP_MAP) && !SSticker.HasRoundStarted())
```
this code is highly non generic and a bit confusing, please make it proper

Lumipharon replied:
less guh'd, let me know if it's still scuff though.

Lumipharon commented:
Patrol bases have been updated, thanks to Shelteredfreak for making them.

Lumipharon commented:
Bioscan added, triggering every 5 minutes.

This should help in situations where nothing is happening because you don't know where the other side is. i.e. both teams round start turtling, or when there's like one dude left on one side.

Lumipharon commented:
SG loadouts tweaked, added a sword loadout because why not, minor tweaks to SOM armor, and some general name/desc cleanup.

RipGrayson reviewed (approved):
Mapping side of this PR looks fine.

RipGrayson reviewed (changes requested):
Actually, doing a more thorough review looking for varedits I found this on the protein bars.
![badbars](https://user-images.githubusercontent.com/49290523/180334735-1f0a817b-e5f1-48bc-9356-826c84148fe4.png)
This isn't a proper desc, it isn't even a proper string. Please fix it.

XSlayer300 commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -0,0 +1,265 @@
+	var/som_scan_name = "Long Range Tactical Bioscan Status"
+	var/som_scan_input = {"Bioscan complete.
+
+Sensors indicate [numTGMCr || "no"] unknown lifeform signature[numTGMCr > 1 ? "s":""] present in the area of operation[TGMCLocation ? ", including one at:[TGMCLocation]":""]"}
```
```suggestion
Sensors indicate [numTGMCr || "no"] unknown IFF signature[numTGMCr > 1 ? "s":""] present in the area of operations[TGMCLocation ? ", including one in [TGMCLocation]":""]"}
```
![image](https://user-images.githubusercontent.com/35091844/180349612-bf95112c-ec82-418b-ad6a-9ae98d3ffec1.png)

I still don't understand the gap for this, so... somehow fix this.

Lumipharon replied:
![image](https://user-images.githubusercontent.com/7869430/180351458-c866798b-872f-4487-af00-5293e49d6e45.png)
already fixed, server probably just isn't updated.

XSlayer300 commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -0,0 +1,265 @@
+	var/marine_scan_name = "Long Range Tactical Bioscan Status"
+	var/marine_scan_input = {"Bioscan complete.
+
+Sensors indicate [numSOMr || "no"] unknown lifeform signature[numSOMr > 1 ? "s":""] present in the area of operation[SOMLocation ? ", including one at:[SOMLocation]":""]"}
```
```suggestion
Sensors indicate [numSOMr || "no"] unknown IFF signature[numSOMr > 1 ? "s":""] present in the area of operations[SOMLocation ? ", including one in [SOMLocation]":""]"}
```

Lumipharon replied:
Good catch on operations. 
"At" is used else where and both make grammatical sense, so I'll leave it as is for consistancy. Fixed the missing space though.
Lifeform I also want to keep because it's generic, and consistant with bioscan in HvX because xenos sure don't have IFF. Also the name bioscan implies it's just picking up lifesigns somehow.

Lumipharon commented:
Minor fixes, replace kits in first aid pouches with biard/kelo injectors. I'm still trying to keep the amount of meds down however, so not pill packets.

Lumipharon commented:
Adds a wave respawning mechanic. 15 minutes are shutter drop, then every 10 minutes after that, all dead players will get their respawn timers cleared, allowing them to respawn.

Note that in HvH, 
A: Going perma frees your job slot
B: You cannot change faction 

So everyone that's dead should be able to respawn, but in some very edge cases someone might miss out on actually being able to spawn. Namely if the players on the other side choose not to respawn, and your side suddenly outnumbers the enemy side by a significant margin, since one side cannot outnumber the other by more than 20%, you will not have a valid faction to join.

Lumipharon commented:
Veteran slots now scale by pop. I may need to tweak this to find the right number, but for now it's 2 slots base, then another slot for every 5 dudes that join (excluding further vets).

Note that as above, going perma frees your slow, so everyone respawning will in fact free existing vet slots as well as ultimately open new ones. However loadout specific quantities do NOT renew, so you don't get more culverins (I also added a limit on caliver+ backpack loadout).

I'm going to remove charger loadouts from medic and engie, but I'll do that later today.

Lumipharon commented:
Final set of changes, I'll save anything else for future PR's.
Various loadout tweaks and additions, notably the loss of chargers for SOM medics and engies
Grenade launchers (including UGL) can now fire all the new HvH specific nades
Added some mag harnesses and toolbelts to the patrol bases (so you can crowbar doors when you have no engineers, etc)
Fixes to the deflagrate proc: less likely to trigger (especially at long range), but the soft cc will always activate as intended.

Lumipharon commented:
I lied, last change. Fixed bioscan so it only checks the ground map. Also cleaned up the code for count_humans, and by extension bioscan + check_finished

TiviPlus commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -117,23 +117,23 @@
-	var/TGMCLocation = locations[FACTION_TERRAGOV]
-	var/SOMLocation	= locations[FACTION_SOM]
+	var/numTGMCr = BIOSCAN_DELTA(num_tgmc, delta)
+	var/numSOMr = BIOSCAN_DELTA(num_som, delta)
```
No mixed case

Lumipharon replied:
done

RipGrayson reviewed (approved):
Mapping part of this PR is fine again

BraveMole commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -0,0 +1,266 @@
+	///The length of time until round ends.
+	var/max_game_time = 35 MINUTES
+	///Whether the max game time has been reached
+	var/max_time_reached
```
```suggestion
	var/max_time_reached = FALSE
```

BraveMole commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -0,0 +1,266 @@
+	else
+		return "Patrol finished"
+
+/datum/game_mode/combat_patrol/proc/set_game_end()
```
we have var callback btw, but doesn't matter here

BraveMole commented on code/datums/gamemodes/combat_patrol.dm:
```diff
@@ -0,0 +1,266 @@
+			continue
+		if(H.faction == FACTION_SOM)
+			som_dead += H
+		if(H.faction == FACTION_TERRAGOV)
```
else if

Lumipharon replied:
done

BraveMole commented on code/game/data_huds.dm:
```diff
@@ -510,13 +510,21 @@
+	var/hud_type
+	if(faction == FACTION_TERRAGOV)
+		hud_type = SQUAD_HUD_TERRAGOV
+	if(faction == FACTION_TERRAGOV_REBEL)
```
else if

Lumipharon replied:
done

BraveMole commented on code/game/data_huds.dm:
```diff
@@ -510,13 +510,21 @@
+		hud_type = SQUAD_HUD_TERRAGOV
+	if(faction == FACTION_TERRAGOV_REBEL)
+		hud_type = SQUAD_HUD_REBEL
+	if(faction == FACTION_SOM)
```
else if

Lumipharon replied:
done

BraveMole commented on code/game/objects/effects/landmarks/landmarks.dm:
```diff
@@ -460,3 +465,14 @@
+	//ID to link with an associated start point
+	var/id = null
+
+/obj/effect/landmark/patrol_point/Initialize()
```
need a destructor

Lumipharon replied:
done

BraveMole commented on code/game/objects/effects/temporary_visuals/miscellaneous.dm:
```diff
@@ -89,14 +89,20 @@
+	var/hud_type
+	if(faction == FACTION_TERRAGOV)
+		hud_type = DATA_HUD_SQUAD_TERRAGOV
+	if(faction == FACTION_TERRAGOV_REBEL)
```
else if

Lumipharon replied:
done

BraveMole commented on code/game/objects/effects/temporary_visuals/miscellaneous.dm:
```diff
@@ -89,14 +89,20 @@
+		hud_type = DATA_HUD_SQUAD_TERRAGOV
+	if(faction == FACTION_TERRAGOV_REBEL)
+		hud_type = DATA_HUD_SQUAD_REBEL
+	if(faction == FACTION_SOM)
```
else if

Lumipharon replied:
done

BraveMole commented on code/game/objects/effects/temporary_visuals/miscellaneous.dm:
```diff
@@ -120,7 +126,13 @@
+	var/hud_type
+	if(faction == FACTION_TERRAGOV)
+		hud_type = DATA_HUD_SQUAD_TERRAGOV
+	if(faction == FACTION_TERRAGOV_REBEL)
```
ditto

Lumipharon replied:
done

BraveMole commented on code/game/objects/items/radio/headset.dm:
```diff
@@ -271,7 +272,12 @@ GLOBAL_LIST_INIT(channel_tokens, list(
-		marker_flags = hud_type == DATA_HUD_SQUAD_TERRAGOV ? MINIMAP_FLAG_MARINE : MINIMAP_FLAG_MARINE_REBEL
+		if(hud_type == DATA_HUD_SQUAD_TERRAGOV)
+			marker_flags = MINIMAP_FLAG_MARINE
+		if(hud_type == DATA_HUD_SQUAD_REBEL)
```
else if

Lumipharon replied:
done

BraveMole commented on code/game/objects/items/radio/headset.dm:
```diff
@@ -289,7 +295,13 @@ GLOBAL_LIST_INIT(channel_tokens, list(
+	var/marker_flags
+	if(hud_type == DATA_HUD_SQUAD_TERRAGOV)
+		marker_flags = MINIMAP_FLAG_MARINE
+	if(hud_type == DATA_HUD_SQUAD_REBEL)
```
else if

Lumipharon replied:
done

BraveMole commented on code/game/objects/items/radio/headset.dm:
```diff
@@ -298,7 +310,13 @@ GLOBAL_LIST_INIT(channel_tokens, list(
+	var/marker_flags
+	if(hud_type == DATA_HUD_SQUAD_TERRAGOV)
+		marker_flags = MINIMAP_FLAG_MARINE
+	if(hud_type == DATA_HUD_SQUAD_REBEL)
```
else if

Lumipharon replied:
done

BraveMole commented on code/game/objects/structures/patrol_points.dm:
```diff
@@ -0,0 +1,51 @@
+
+
+/obj/structure/patrol_point/LateInitialize()
+	. = ..()
```
```suggestion
```

BraveMole commented on code/game/objects/structures/patrol_points.dm:
```diff
@@ -0,0 +1,51 @@
+/obj/structure/patrol_point/proc/create_link()
+	for(var/obj/effect/landmark/patrol_point/exit_point AS in GLOB.patrol_point_list)
+		if(exit_point.id == id)
+			linked_point = exit_point
```
```suggestion
			linked_point = exit_point
			return
```

BraveMole commented on code/modules/projectiles/ammo_datums.dm:
```diff
@@ -1342,6 +1353,188 @@ datum/ammo/bullet/revolver/tp44
+	shrapnel_chance = 0
+	max_range = 6
+	bullet_color = COLOR_VERY_SOFT_YELLOW
+	var/datum/effect_system/smoke_spread/smoketype = /datum/effect_system/smoke_spread
```
doc

Lumipharon replied:
done

Lumipharon commented on code/__DEFINES/mode.dm:
```diff
@@ -69,6 +71,12 @@
+#define MODE_SOM_MARINE_MINOR "Marine Minor Victory"
+#define MODE_SOM_SOM_MAJOR "Sons of Mars Major Victory"
+#define MODE_SOM_SOM_MINOR "Sons of Mars Minor Victory"
+#define MODE_COMBAT_PATROL_DRAW "DRAW: Mutual Annihilation"
```
done

Lumipharon commented:
Fixed a funny bug in situations where no one votes at round end.
